docs(how-to): add IFPS doc MTA-3327 (#1857)
scaleway/docs-content: labs/ipfs/how-to/add-a-pin.mdx

---
meta:
  title: How to add a pin to your data and retrieve it
  description: This page explains how to add a pin to your data and retrieve it using Scaleway IPFS Pinning
content:
  h1: How to add a pin to your data and retrieve it
  paragraph: This page explains how to add a pin to your data and retrieve it using Scaleway IPFS Pinning
tags: ipfs-pinning ipfs-storage ipfs pinning storage labs web3 volume pin
dates:
  validation: 2023-06-27
  posted: 2023-06-27
categories:
  - labs
---

> Scaleway IPFS pinning allows you to permanently store a copy of your data from the public IPFS network on a Scaleway-owned node. That way, your data remains available and accessible even when your local machine is offline.

Reviewer: Is the product name "IPFS" or "IPFS Pinning"? If it is the latter, I think the "P" in "pinning" should be capitalized :)

<Message type="requirement">

>   - You have an account and are logged into the [Scaleway Console](https://console.scaleway.com/)

Reviewer: ```suggestion
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com/)
```

  - You have downloaded and installed [IPFS Desktop App](https://docs.ipfs.tech/install/ipfs-desktop/)

>   - You have added data to your own local node and retrieved its CID

Reviewer: Maybe add a link to a CID concept or something to explain what it is

</Message>

## How to pin your data

> 1. Click **IPFS Pinning** in the **Labs** section of the [Scaleway Console](https://console.scaleway.com) side menu. The IPFS Pinning section displays.

Reviewer: ```suggestion
1. Click **IPFS Pinning** in the **Labs** section of the [Scaleway console](https://console.scaleway.com) side menu. The **IPFS Pinning** section displays.
```

2. Click the <Icon name="seeMore" /> icon next to your desired volume. The **Pins** section displays.
3. Click **Add Pin** to add a pin to your data. A pop-up displays, prompting you to enter the data's information.
4. Enter the [CID](/en/docs/labs/ipfs/concepts/#content-identifier-(cid)) of your data as well as a name for it.
5. Click **Submit**. The CID and name of your data appears under the **Pins** section. 

## How to retrieve your data 

> Now that you have pinned your data, you can retrieve it using [CloudFlare](https://www.cloudflare.com/), [Pinata](https://www.pinata.cloud/) or [Protocol Labs](https://protocol.ai/). These are IPFS gateways, which are services that allow you to interact with the IPFS network using regular, HTTP/HTTPS web protocols. This means you don't need a particular IPFS software to retrieve your data - you can instead use your regular web browser.

Reviewer: ```suggestion
Now that you have pinned your data, you can retrieve it using [CloudFlare](https://www.cloudflare.com/), [Pinata](https://www.pinata.cloud/) or [Protocol Labs](https://protocol.ai/). These are IPFS gateways, which are services that allow you to interact with the IPFS network using regular, HTTP/HTTPS web protocols. This means you do not need a particular IPFS software to retrieve your data - you can instead use your regular web browser.
```
